server/src/processing/LinkDetectionModule/LinkDetectionModule.ts

```diff
@@ -32,7 +33,13 @@ export class LinkDetectionModule extends Module {
   public static moduleName = 'link-detection';
 
   public async main(doc: Document): Promise<Document> {
+    const fileType: { ext: string; mime: string } = filetype(fs.readFileSync(doc.inputFile));
+    if (fileType === null || fileType.ext !== 'pdf') {
+      logger.warn(`Warning: The input file ${doc.inputFile} is not a PDF (${utils.prettifyObject(fileType)}); not fetching meta information for link detection..`);
+      return undefined;
```

Comment:
shouldn't this be `return doc`?

Reply:
Also, please keep in mind that this module not only gets links from metadata, but on Words also! And with this condition, It will not get textual links from non-pdf files.

Reply:
You're right! Sorted it out in 7fbfdd9